### PR TITLE
Fix #2922: file uploads fail to render curl command

### LINKED ISSUE #2922: Post requests w/ multipart form data fail to render curl command
Rendering the curl command no longer works for POST requests that have formData. The request body isn't getting set as a string before being passed to `curlify`. As a result, line `21` in `src/core/curlify.js` throws the following error when you Execute the request: 

`TypeError: request.get(...).split is not a function`

And you see this in the UI: `😱 Could not render Curl, see the console.`

The line in question:

`let formDataBody = request.get("body").split("&")`

Found in swagger-ui version: `3.0.6`

Here is my paired down `swagger.json` that reproduces the issue:
```json
{
  "swagger": "2.0",
  "info": {
    "version": "1.0.0",
    "title": "NA",
    "description": "..."
  },
  "basePath": "/api",
  "tags": [
    {
      "name": "products",
      "description": "Product APIs"
    },
    {
      "name": "scans",
      "description": "Scanning Files"
    }
  ],
  "schemes": [
    "http",
    "https"
  ],
  "paths": {
    "/scans": {
      "post": {
        "description": "Upload a file to be scanned without a product association. Use\nthis for 'quick scans' which are one-off scans whose results are\nuncorrelated with previous ones.\n",
        "tags": [
          "scans"
        ],
        "consumes": [
          "multipart/form-data"
        ],
        "produces": [
          "application/json"
        ],
        "parameters": [
          {
            "name": "file",
            "in": "formData",
            "type": "file",
            "description": "The file to scan.",
            "required": true
          }
        ],
        "responses": {
          "201": {
            "description": "The file was successfully uploaded and the scan has started.\nReturns a scan object containing the status of the scan. Use\nthe returned ID to poll /scans/{scanId} to monitor the progress\nof the scan.\n",
            "schema": {
              "$ref": "#/definitions/Scan"
            }
          },
          "default": {
            "description": "Generic error response",
            "schema": {
              "$ref": "#/definitions/Error"
            }
          }
        }
      }
    }
  },
  "definitions": {
    "Error": {
      "type": "object",
      "required": [
        "message"
      ],
      "properties": {
        "message": {
          "type": "string"
        }
      }
    },
    "Scan": {
      "type": "object",
      "required": [
        "id",
        "recipe",
        "progress",
        "uploaded_on"
      ],
      "properties": {
        "id": {
          "type": "string",
          "description": "Scan ID"
        },
        "recipe": {
          "type": "string",
          "description": "The type of scan. This varies depending on the type of file uploaded."
        },
        "progress": {
          "type": "array",
          "description": "A list of all the steps in the scan.",
          "items": {
            "$ref": "#/definitions/ScanProgress"
          }
        },
        "created_on": {
          "type": "integer",
          "format": "int64",
          "description": "The time the file upload began."
        },
        "uploaded_on": {
          "type": "number",
          "format": "int64",
          "description": "The time the file upload finished."
        },
        "started_on": {
          "type": "number",
          "format": "int64",
          "description": "The time the scan was started. Omitted if the scan has not yet started."
        },
        "completed_on": {
          "type": "number",
          "format": "int64",
          "description": "The time the scan was completed. Omitted if the scan is not complete."
        },
        "file": {
          "$ref": "#/definitions/File"
        }
      }
    },
    "ScanProgress": {
      "type": "object",
      "required": [
        "name",
        "status",
        "started_on"
      ],
      "properties": {
        "name": {
          "type": "string",
          "description": "Name of the scan step."
        },
        "status": {
          "type": "string",
          "description": "The status of the scan step.",
          "enum": [
            "pending",
            "in progress",
            "complete",
            "failed"
          ]
        },
        "started_on": {
          "type": "string",
          "format": "date-time",
          "description": "The time the scan step was started."
        },
        "completed_on": {
          "type": "string",
          "format": "date-time",
          "description": "The time the scan step was completed. This is omitted if the scan step is not complete."
        },
        "message": {
          "type": "string",
          "description": "An optional message explaining the status. For instance if there was an error this would explain the cause."
        }
      }
    },
    "File": {
      "type": "object",
      "required": [
        "name"
      ],
      "properties": {
        "id": {
          "type": "string",
          "description": "Filename id as a UUID."
        },
        "name": {
          "type": "string",
          "description": "The scan filename."
        },
        "size": {
          "type": "number",
          "format": "int64",
          "description": "The size of the file."
        },
        "scantype": {
          "type": "string",
          "description": "The type of file as described by the system."
        }
      }
    }
  }
}
```

### PR DESCRIPTION
**Pull request by dwickern**
The request body is an immutable.js OrderedMap rather than a string, so this fixes the type error:

```
  TypeError: request.get(...).split is not a function
```

Adds a special curl command argument for files, e.g.

```
  curl -F "param=value" -F "file=@filename.txt;type=text/plain"
```

### CODE REVIEW

**Comment by dwickern** [comment]
@shockey Agreed, I wanted to see the CI results first.

Since curlify depends on `window.File` now I'll have to mock one up for the test context.

File: test/core/curlify.js [hunk 1]
```diff
@@ -131,12 +132,35 @@ describe("curlify", function() {
             url: "http://example.com",
             method: "POST",
             headers: { "content-type": "multipart/form-data" },
-            body: "id=123&name=Sahar"
+            body: [
```
Comment (shockey) [inline test/core/curlify.js]: Just wondering - why was this changed?

Comment (shockey) [inline test/core/curlify.js]: Just wondering - why was this changed?

Comment (dwickern) [inline test/core/curlify.js]: I changed it to match the types which are actually passed from the UI.

Judging from the old code in swagger-ui 2.x the multipart form body was passed as a string.

In 3.x a multipart form body gets passed as this kind of structure:
```
Immutable.OrderedMap(
 1 -> Immutable.List("param1", "value1"),
 2 -> Immutable.List("param2", "value2"),
 ... and so on
)
```
I didn't want to change structure and risk breaking some other UI somewhere.

**Comment by shockey** [comment]
Thanks for the contribution (and the tests!), @dwickern 😄

**Comment by dwickern** [comment]
@shockey Thank you for your time and effort maintaining this 😃

**Pull request merged**